duviz.py

```diff
@@ -210,6 +215,76 @@ def pairs(listing: str) -> Iterator[Tuple[List[str], int]]:
         return tree
 
 
+class ZipListingParseException(ParseException):
+    pass
+
+
+class ZipListingParser:
+    def list_and_parse(self, zip_path: str) -> SizeTree:
+        """
+        Run `unzip -l` on a ZIP file and parse the listing to a `SizeTree`
+        :param zip_path:  ZIP file path
+        :return: size tree
+        """
+        command = ["unzip", "-l", zip_path]
```

Comment:
instead of using subprocess `unzip -l`,  read  zip file directly with `zipfile` module